doc/man3/EVP_PKEY_set1_RSA.pod

```diff
@@ -103,6 +102,14 @@ EVP_PKEY_assign_RSA(), EVP_PKEY_assign_DSA(), EVP_PKEY_assign_DH(),
 EVP_PKEY_assign_EC_KEY(), EVP_PKEY_assign_POLY1305()
 and EVP_PKEY_assign_SIPHASH() are implemented as macros.
 
+EVP_PKEY_assign_EC_KEY() looks at the curve name id to determine if
+the passed B<EC_KEY> is an L<SM2(7)> key, and will set the B<EVP_PKEY>
+type to B<EVP_PKEY_SM2> in that case, instead of B<EVP_PKEY_EC>.
+This B<EVP_PKEY> type can be switched to B<EVP_PKEY_EC> with a call to
+EVP_PKEY_set_alias_type() if it's desirable to do a normal EC
+computations with the SM2 curve instead of the special SM2
+computations.
```

Comment:
And vice versa...i.e. you can take a key for a non-SM2 curve and convert it to SM2.

Reply:
Yes. You can see it happen in full action in ecdsatest. Basically, you can call `EVP_PKEY_set_alias_type` at any type with the desired key type, `EVP_PKEY_EC` or `EVP_PKEY_SM2`, and it will work, the next operation will select a pmeth according to that type

Reply:
Heh, I discover that I'm repeating myself...  ah well :wink:

Reply:
Done